### PR TITLE
US136416: Add Hindi support `hi-in`

### PR DESCRIPTION
**Pull request by mostlyfabulous**
[US136416](https://rally1.rallydev.com/#/?detail=/userstory/627390519205&fdp=true): [Activity Feed] Support Hindi (hi-IN)
Ref: https://github.com/Brightspace/serge-localize/blob/master/static/fra.common.serge.inc#L27

### CODE REVIEW

**Review by mdulat — APPROVED** [review]

**Review by bearfriend — APPROVED** [review]

File: helpers/localization.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { getLocalizeOverrideResources } from '@brightspace-ui/core/helpers/getLocalizeResources.js';
 import { resolveUrl } from '@polymer/polymer/lib/utils/resolve-url.js';
 
-const SUPPORTED_LANGUAGES = ['ar-sa', 'ar', 'da-dk', 'da', 'de-de', 'de', 'en-ca', 'en-gb', 'en-us', 'en',
-	'es-mx', 'es', 'fi-fl', 'fi', 'fr-ON', 'fr-ca', 'fr-fr', 'fr', 'ja-jp', 'ja', 'ko-kr', 'ko', 'nb-no', 'nb',
-	'nl-nl', 'nl', 'pt-br', 'pt', 'sv-se', 'sv', 'tr-tr', 'tr', 'zh-cn', 'zh-tw', 'zh'];
+const SUPPORTED_LANGUAGES = ['ar', 'cy', 'da', 'de', 'en', 'es-es', 'es', 'fi-fl', 'fi',
```
Comment (mostlyfabulous) [inline helpers/localization.js:4]: I'm unsure it it was okay to remove `en-ca`, en-gb` and `en-us`. Also, I'll need to find out why we write it as 'fr-ON' here.

**Comment by mostlyfabulous** [comment]
I renamed the `cy-gb` file to be `cy` so that it matches the output. Is this okay?

File: helpers/localization.js (hunk 1 above)
Comment (bearfriend) [reply helpers/localization.js:4]: It _should_ be ok to remove all those, and my guess is it should be `fr-on`, but yeah, definitely just do some manual testing. Maybe set a breakpoint in here and try overwriting the `langs` array that you get in `getLocalizeResources` and see how it resolves.

It also may just be easier to try switching to the [`LocalizeDynamicMixin`](https://github.com/BrightspaceUI/core/blob/main/mixins/localize-mixin.md#example-2-dynamically-imported-resources).

**Comment by bearfriend** [comment]
`zh` should also be renamed to `zh-cn`

File: helpers/localization.js (hunk 1 above)
Comment (mostlyfabulous) [reply helpers/localization.js:4]: Just prototyping that out now

File: mixins/base-mixin.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
 
-const baseMixinUrl = import.meta.url;
-
 export const BaseMixin = superclass =>
-	class extends LocalizeMixin(RtlMixin(superclass)) {
-		static async getLocalizeResources(langs) {
-			return getLocalizeResources(langs, baseMixinUrl);
-		}
+	class extends LocalizeDynamicMixin(RtlMixin(superclass)) {
```
Comment (mostlyfabulous) [inline mixins/base-mixin.js:6]: @mdgbayly I'm upgrading the localization mixin, anything to bear in mind?

Comment (mostlyfabulous) [reply mixins/base-mixin.js:6]: Resolved in Slack

File: mixins/base-mixin.js [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
-
 export const BaseMixin = superclass =>
-	class extends LocalizeMixin(RtlMixin(superclass)) {
-		static async getLocalizeResources(langs) {
-			return getLocalizeResources(langs, baseMixinUrl);
+	class extends LocalizeDynamicMixin(RtlMixin(superclass)) {
+		static get localizeConfig() {
+			return {
+				// Import path must be relative
+				importFunc: async lang => (await import(`../locales/${lang}.js`)).default,
```
Comment (mostlyfabulous) [inline mixins/base-mixin.js:10]: Curly braces triggers ambiguous error:
```
TypeError: Cannot read property 'range' of null
Occurred while linting /home/runner/work/attachment/attachment/mixins/base-mixin.js:1
    at OffsetStorage.setDesiredOffset (/home/runner/work/attachment/attachment/node_modules/eslint/lib/rules/indent.js:340:45)
    at /home/runner/work/attachment/attachment/node_modules/eslint/lib/rules/indent.js:[13](https://github.com/Brightspace/attachment/runs/5396011856?check_suite_focus=true#step:5:13)58:29
    at Array.forEach (<anonymous>)
    at Object.TemplateLiteral [as listener] (/home/runner/work/attachment/attachment/node_modules/eslint/lib/rules/indent.js:1350:34)
    at /home/runner/work/attachment/attachment/node_modules/eslint/lib/rules/indent.js:[15](https://github.com/Brightspace/attachment/runs/5396011856?check_suite_focus=true#step:5:15)96:55
    at Array.forEach (<anonymous>)
    at Program:exit (/home/runner/work/attachment/attachment/node_modules/eslint/lib/rules/indent.js:1596:26)
    at /home/runner/work/attachment/attachment/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/runner/work/attachment/attachment/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
```

**Comment by mostlyfabulous** [comment]
image
Can confirm the specific `fr-on` term for Dropbox `Tâche` loads using storybook

File: mixins/base-mixin.js [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
+				// Optionally enable OSLO
+				osloCollection: '@d2l\\d2l-attachment\\d2l-attachment',
```
Comment (mostlyfabulous) [inline mixins/base-mixin.js:12]: @stevenao does this look correct?

Comment (mostlyfabulous) [inline mixins/base-mixin.js:12]: @kieranderson do you know if this remains the same? 
Original: https://github.com/Brightspace/attachment/blob/master/helpers/localization.js#L17

File: locales/fi-fl.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export default {
```
Comment (bearfriend) [inline locales/fi-fl.js:1]: This probably isn't doing anything but maybe blocking fallback to `fi`.

Comment (mostlyfabulous) [reply locales/fi-fl.js:1]: Will remove, thanks

File: mixins/base-mixin.js (hunk 4 above)
Comment (kieranderson) [reply mixins/base-mixin.js:12]: Same is good. :+1:

**Pull request merged**